src/neuroconv/tools/roiextractors/roiextractors.py

```diff
@@ -346,18 +370,18 @@ def add_photon_series(
         ), "Received metadata for 'OnePhotonSeries' but `photon_series_type` was not explicitly specified."
 
     # Tests if TwoPhotonSeries//OnePhotonSeries already exists in acquisition
-    photon_series_kwargs = metadata_copy["Ophys"][photon_series_type][photon_series_index]
-    photon_series_name = photon_series_kwargs["name"]
+    photon_series_metadata = metadata_copy["Ophys"][photon_series_type][photon_series_index]
+    photon_series_name = photon_series_metadata["name"]
 
     if photon_series_name in nwbfile.acquisition:
         warn(f"{photon_series_name} already on nwbfile")
         return nwbfile
 
     # Add the image plane to nwb
-    # TODO: change imaging_plane_index to photon_series_key
-    nwbfile = add_imaging_plane(nwbfile=nwbfile, metadata=metadata_copy, imaging_plane_index=photon_series_index)
-    imaging_plane_name = photon_series_kwargs["imaging_plane"]
+    imaging_plane_name = photon_series_metadata["imaging_plane"]
+    add_imaging_plane(nwbfile=nwbfile, metadata=metadata_copy, imaging_plane_name=imaging_plane_name)
     imaging_plane = nwbfile.get_imaging_plane(name=imaging_plane_name)
+    photon_series_kwargs = deepcopy(photon_series_metadata)
     photon_series_kwargs.update(imaging_plane=imaging_plane)
```

Comment:
> My memory is that we needed to deepcopy the metadata so we don't modify the data that the user passes but the hpoton_series_metadata is already deepcopied so why copy it again?

You're right, the reason why I chose this is to avoid overwriting the "imaging_plane" in the metadata with the actual ImagingPlane object at this line.

Reply:
Ah, OK, probably not necessary anymore then.